src/main/java/edu/ksu/canvas/interfaces/SelectiveDataReader.java

```diff
@@ -0,0 +1,17 @@
+package edu.ksu.canvas.interfaces;
+
+import edu.ksu.canvas.model.SelectiveData;
+import edu.ksu.canvas.requestOptions.GetSelectiveDataOptions;
+
+import java.io.IOException;
+import java.util.List;
+
+public interface SelectiveDataReader extends CanvasReader<SelectiveData, SelectiveDataReader> {
+    /**
+     * Get selective data of a migration.
+     * @param options The destination item ID for the migration, the migration id and the type
+     * @return List of selectable items for the migration
+     * @throws IOException When there is an error communicating with Canvas
+     */
+    List<SelectiveData> getCourseSelectiveDataFromMigration(GetSelectiveDataOptions options) throws IOException;
```

Comment:
Using the optional type parameter in the GetSelectiveDataOptions, I'll add the JavaDoc in that constructor as you said and hopefully it will be clearer.

Reply:
But don't you need to make an additional GET request to the URL returned in the sub_items?

Reply:
Yes, but the url is the same just with an additional type parameter:

https://ox.instructure.com/api/v1/courses/3758/content_migrations/19/selective_data

then

https://ox.instructure.com/api/v1/courses/3758/content_migrations/19/selective_data?type=assignments

The endpoint is the same, it's just the options object that changes.

Reply:
There's no guarantee that that will be the URL though, and then you have to parse the URL and extract out the data which is error prone.

Reply:
After discussion, added comment in the code:

>      * NOTE : Urls for subitems are expected to be on the same endpoint as the top-level but adding a type parameter
>      *        The available types can be retrieved in the top-level GET, then a type can be added to the GetSelectiveDataOptions object on the following call